Libraries/LibCore/EventLoopImplementation.cpp

```diff
@@ -26,7 +26,8 @@ EventLoopImplementation::~EventLoopImplementation() = default;
 void EventLoopImplementation::deferred_invoke(Function<void()>&& invokee)
 {
     m_thread_event_queue.deferred_invoke(move(invokee));
-    wake();
+    if (&m_thread_event_queue != ThreadEventQueue::current_or_null())
+        wake();
```

Comment:
I have not looked deeply in the event loop before, but is it guaranteed that we don't hit poll again before running any of the deferred_invoke tasks?

Since otherwise I suppose we might also deadlock no other fd's in the poll loop have any event on them before we hit the poll loop, and the deferred tasks are not run by that point we get stuck at this point: https://github.com/LadybirdBrowser/ladybird/blob/89dbdd3411a3037ec31dcd96c15a47f062c7f48f/Libraries/LibCore/EventLoopImplementationUnix.cpp#L353 not ever running the deferred tasks

Seems like we might hit that if you deferred_invoke and then immediately pump: https://github.com/LadybirdBrowser/ladybird/blob/89dbdd3411a3037ec31dcd96c15a47f062c7f48f/Libraries/LibCore/EventLoopImplementationUnix.cpp#L307

Reply:
Right, we should end up with a timeout of zero and process the deferred functions almost immediately.